ott/core/continuous_barycenter.py

```diff
@@ -175,6 +175,12 @@ def init_state(
     if x_init is not None:
       assert bar_size == x_init.shape[0]
       x = x_init
+    elif hasattr(bar_prob.cost_fn, 'barycenter_init'):
```

Comment:
Yes I agree.

The need for appropriate initialization came up with the test ``test_bures_barycenter`` in ``continuous_barycenter.py`` where because of the symmetry of the problem the random initialization you have proposed (randomly sampling points from the measures averaged) led to unexpected results. As a reminder: the test would fail if the user did not pass a custom initialization for the barycenter.

Now, with this initialization, the barycenter given appropriate results even if the user does not pass an appropriate initialization. I have modified ``test_bures_barycenter`` to illustrate this.

I agree with you that it would be nicer to have a ``barycenter_init`` for all costs. especially since this behavior can happen also with the Euclidean cost in case of symmetry (the means of the Gaussians with Bures will sort of have similar behavior with points and Euclidean cost). However, since the initialization I suggested here for the means in the case of Bures has the same complexity as one barycenter iteration for Euclidean it seemed as a bit of an overkill.

The tests for ``jit=True`` in continuous barycenter pass and I am using by default the ``barycenter_init``, so it seems to be alright. Can look into this further.

Reply:
I set the ``barycenter_init`` (with the current random initialization) as a method of ``CostFn``, so it is inherited by all costs. For now only overridden by Bures.

The ``hassatr`` was fine ``jit``-wise because it does not check the value of any tracer.